catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityFilterPlugin.java

```diff
@@ -104,17 +108,20 @@ public PolicyResponse processPostQuery(Result input, Map<String, Serializable> p
         Metacard metacard = input.getMetacard();
         HashMap<String, Set<String>> securityMap = new HashMap<>();
 
-        if ((metacard.getAttribute(Validation.VALIDATION_ERRORS) != null && metacard.getAttribute(
-                Validation.VALIDATION_ERRORS)
-                .getValues() != null) || (
-                metacard.getAttribute(Validation.VALIDATION_WARNINGS) != null &&
-                        metacard.getAttribute(Validation.VALIDATION_WARNINGS)
-                                .getValues() != null)) {
+        if (filterErrors && metacard.getAttribute(Validation.VALIDATION_ERRORS) != null &&
+                metacard.getAttribute(Validation.VALIDATION_ERRORS)
+                        .getValues() != null) {
+            for (Map.Entry<String, List<String>> attributeMapping : attributeMap.entrySet()) {
```

Comment:
I'm confused what's going on here. Maybe I'm just not seeing something but your if and else if conditions, while different, do the same thing. You loop over the same `attributeMap` and do the exact same puts to the `securityMap`.

Do I just not see some subtle difference between the blocks?